OpenGDPR_specification.md

```diff
@@ -0,0 +1,723 @@
+
+- Defining the technical measures to describe the fulfill of Data Subject requests.
+It is the responsibility of each Data Controller and Data Processor to interpret and
+apply the GDPR to honor Data Subject requests (Chapter 3).
+- The protocol for communications between Controllers and Data Subjects.
+- The protocol for communications between Controllers or Processors and Supervisory
+Authorities.
+- The protocol for communication of the results of an access or portability request.
+
+### 1.1.  Notational Conventions
+- The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119].
+- Global Unique Identifiers (GUID) **MUST** be lowercase and v4 format.
+
+
+## 2. Terms and Definitions
+
+#### Data Subject Request
+
+A request from a Data Subject exercising their Data Subject Rights as defined within the GDPR under Chapter 3.
+
+#### Fulfillment
+
+Enacting compliance related activities to honor an OpenGDPR request.
+
+## 3. OpenGDPR Basics
+
+### 3.1. Roles and Responsibilities
+
+#### Data Subject
+ 
+  A European Union resident whose personal data is being processed.
+
+#### Data Controller
+  
+  An entity which makes the decision about what personal data  will be processed and the types of processing that will be done with respect to that personal data. The Data Controller receives Data Subject requests from the Data Subjects and validates them. The Data Controller **SHOULD** provide a callback endpoint. The Data Controller **SHOULD** verify response signatures. Referenced as "Controller."
+
+#### Data Processor
+
+  The organization that processes data pursuant to the instructions of the Controller on behalf of the Controller. The Data Processor receives data subject requests via RESTful endpoints and is responsible for fulfilling requests. The Data Processor **MUST** provide a signed response to requests. The Data Processor **MUST** honor callbacks.  Data Processors **MUST** honor callbacks included in requests. 
+
+  Processors **MUST** provide the following endpoints: 
+
+  - `/discovery`
+  - `/status`
+  - `/opengdpr_requests`
+
+### 3.2.  Protocol Flow
+
+  ```
+
+  +-----+        +-----+         +-----+
+  |     |        |     |         |     |
+  |     | -(A)-> |     | -(B)->  |     |
+  |     |        |     |         |     | -(C)+
+  |     |        |     |         |     |     |
+  |     |        |     |         |     | <---+
+  |     |        |     |         |     |
+  |     |        |     |         |     |
+  |     |        |     |         |     |
+  |     |        |     |         |     |
+  |     | <-(E)- |     | <--(D)- |     |
+  +-----+        +-----+         +-----+
+  Data          Controller      Processor
+  Subject
+  ```
+  *Figure 1. Request Sequence Flow*
+
+The flow illustrated in figure 1 includes the following steps:
+
+1. A new data subject request submitted to the Controller
```

Comment:
Should use letters to match the diagram

Reply:
markdown doesn't support letters :( i think i can just manually tap it out though.